### PR TITLE
Cancel websocket timer on failure

### PR DESCRIPTION
**Pull request by miguelportilla**
This fixes a problem where the timeout timer would not be canceled with some errors.

### CODE REVIEW

**Comment by ximinez** [comment]
If you put `fix #2026` in the commit message, the issue will automatically close when the code is merged.

**Comment by vinniefalco** [comment]
Did you intend to submit this pull request against **release** ?

**Comment by nbougalis** [comment]
@vinniefalco: yes - half of this fix is already in develop and we're doing a **0.60.1** since it's affecting some users.

File: src/ripple/server/impl/BaseWSPeer.h [hunk 1]
```diff
@@ -195,7 +203,12 @@ run()
             &BaseWSPeer::run, impl().shared_from_this()));
     impl().ws_.set_option(beast::websocket::decorate(identity{}));
     impl().ws_.set_option(port().pmd_options);
+    impl().ws_.set_option(beast::websocket::ping_callback{
+        std::bind(&BaseWSPeer::on_ping_pong, this,
+            std::placeholders::_1, std::placeholders::_2)});
```
Comment (HowardHinnant) [inline src/ripple/server/impl/BaseWSPeer.h]: Consider lambdas in place of `bind`:
````
[this](auto x, auto const& y) {this->on_ping_pong(x, y);}});
````
Technically the `this->` in the lambda body isn't needed, but gcc won't support that until gcc-7.0.  I tested the above on all three compilers and it appears to work fine.

**Comment by ximinez** [comment]
Merged to `release` as 128f7ce and 8b43d67.

**Pull request closed without merging**